crates/noirc_evaluator/src/ssa/ssa_gen/mod.rs

```diff
@@ -258,6 +290,7 @@ impl<'a> FunctionContext<'a> {
         index: super::ir::value::ValueId,
         element_type: &ast::Type,
         location: Location,
+        max_length: Option<super::ir::value::ValueId>,
```

Comment:
Is this `max_length`? It seems like elsewhere we are tracking the slice's actual length which may just be a witness. Compared to e.g. the `slice + len + capacity` approach we discussed where the capacity was always the maximum length rather than the current length held by `len`.

Reply:
Yeah you are right it is just `length` not the `capacity`. I think I wrote `max_length` because it is what I wanted to check against for accesses, but I will make it just `length`